shop/src/main/java/ro/msg/learning/shop/service/LocationService.java

```diff
@@ -0,0 +1,70 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.mapper.LocationMapper;
+import ro.msg.learning.shop.repository.LocationRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class LocationService {
+
+    public static final String THE_VALUE_IS_ALREADY_IN_THE_LIST = "The value is already in the list.";
+    public static final String ID_INVALID = "ID invalid.";
+    @Autowired
+    private LocationRepository locationRepository;
+    @Autowired
+    private LocationMapper locationMapper;
+
+
+    public Location createLocation(Location location) {
+        return locationRepository.save(location);
+    }
+
+    public List<Location> getAllLocations() {
+        return locationRepository.findAll();
+    }
+
+    public Boolean existById(UUID locationId) {
+        return locationRepository.existsById(locationId);
+    }
+
+    public void deleteById(UUID locationId) {
+        if (Boolean.FALSE.equals(locationRepository.existsById(locationId))) {
+            throw new IllegalArgumentException(ID_INVALID);
```

Comment:
why are you throwing this exception in particular? this can be handled without the exception throwing and catching, especially if the runtime exception can be avoided

Reply:
it was a wrong exception in this case, i changed it